scripts/scil_flip_or_swap_gradients_axis.py

```diff
@@ -0,0 +1,104 @@
+    # will be: parser.error:  argument --final_order: expected one argument
+    p.add_argument('--final_order', metavar='"axis axis axis"',
+                   help="The final order of the axis, compared to original "
+                        "order. \n"
+                        "Choices: ['x', 'y', 'z', '-x', '-y', '-z']\n"
+                        "Ex: to only flip y: --final_order x -y z.\n"
+                        "Ex: to only swap x and y: --final_order y x z.\n"
+                        "Ex: to first flip x, then permute all three axes: "
+                        "--final_order z -x y.")
+
+    gradients_type = p.add_mutually_exclusive_group(required=True)
+    gradients_type.add_argument('--fsl', dest='fsl_bvecs',
+                                action='store_true',
+                                help='Specify fsl format.')
+    gradients_type.add_argument('--mrtrix', dest='fsl_bvecs',
+                                action='store_false',
+                                help='Specify mrtrix format.')
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    # Format final order
+    args.final_order = args.final_order.split()
+    if len(args.final_order) != 3:
+        parser.error("Option --final_order expecting 3 values separated by "
+                     "a space, got: {}".format(args.final_order))
+
+    assert_inputs_exist(parser, args.in_gradient_sampling_file)
+    assert_outputs_exist(parser, args, args.out_gradient_sampling_file)
+
+    # Separating into flip and swap
+    axes_to_flip = []
+    swapped_order = []
+    for axis in args.final_order:
+        if axis[0] == '-':
+            axes_to_flip.append(axis[1])
+            axis = axis[1]
+        swapped_order.append(axis)
+    axes_to_flip = [str_to_index(axis) for axis in axes_to_flip]
+    swapped_order = [str_to_index(axis) for axis in swapped_order]
+
+    # Verifying that user did not ask for, ex, -x x y
+    if len(np.unique(swapped_order)) != 3:
+        parser.error("--final_order should contain the three axis.")
+
+    _, ext = os.path.splitext(args.in_gradient_sampling_file)
+    bvecs = np.loadtxt(args.in_gradient_sampling_file)
+    if args.fsl_bvecs:
+        if ext == '.bvec':
+            bvecs = flip_gradient_sampling(bvecs, axes_to_flip, 'fsl')
+            bvecs = swap_gradient_axis(bvecs, swapped_order, 'fsl')
+            np.savetxt(args.out_gradient_sampling_file, bvecs, "%.8f")
+        else:
+            parser.error('Extension for FSL format should be .bvec.')
+
+    elif ext == '.b':
+        bvecs = flip_gradient_sampling(bvecs, axes_to_flip, 'mrtrix')
+        bvecs = swap_gradient_axis(bvecs, swapped_order, 'mrtrix')
+        np.savetxt(args.out_gradient_sampling_file, bvecs,
+                   "%.8f %.8f %.8f %0.6f")
+    else:
+        parser.error('Extension for MRtrix format should .b.')
```

Comment:
You should have a message for both MRtrix and FSL

Reply:
Done